File: Competitive Programming 3/Section 1.3.3/11559 - Event Planning/Java/Main.java
import java.util.Scanner;

public class Main {
    static Scanner s = new Scanner(System.in);
    public static void main(String[] args){
        while(s.hasNextInt()){
            int n = s.nextInt(), b = s.nextInt(), h = s.nextInt(), w = s.nextInt(),minAns = Integer.MAX_VALUE;
            for(int ih = 0; ih < h ; ih ++){
                int pi = s.nextInt() ; 
                boolean possible = false;
                for(int jw = 0 ; jw < w ; jw++){
                    if(s.nextInt()>=n){
                        possible=true;
                    }
                }
                if(possible && pi*n<=b){
                    minAns = Integer.min(minAns,pi*n);
                }
            }
            System.out.println(minAns==Integer.MAX_VALUE?"stay home":minAns);
        }
    }
}

File: Competitive Programming 3/Section 1.3.3/11559 - Event Planning/C++/main.cpp
#include<iostream>

using namespace std;

int n,b,h,w;

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    while(cin>>n>>b>>h>>w){
        int minAns = b+1;
        for(int ih = 0 ; ih < h ; ih++){
            int pi ; 
            cin >> pi;
            bool possible = false;
            for(int jw = 0 ; jw < w ; jw++){
                int wc;
                cin >> wc;
                if(wc>=n)
                    possible=true;
            }
            if(possible&&pi*n<minAns){
                minAns = pi*n;
            }
        }
        cout << (minAns==b+1?"stay home":to_string(minAns))<<"\n";
    }
    return 0;
}